### PR TITLE
Support sudo users with different capabilities

### PR DESCRIPTION
**Pull request by hkumarmk**
This patch enable to create sudo users with different capabilities, It
support adding hiera data with different set of users with different commands
allowed.

e.g below hiera data will create two sets of users, - admin1 and admin2 with all
access and user1 and user2 with access to commands cat and less.
```
rjil::system::accounts::sudo_users:
  admin:
    users: [admin1,admin2]
  readonly:
    users: [user1,user2]
    commands_allowed: ['/bin/cat','/usr/bin/less']
```

This is required to merge before we can provide engineering people access to read the files in production.

### CODE REVIEW

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
last one failed because one of node had metadata isssues it seems

**Comment by hkumarmk** [comment]
quota exceeded on last one :(

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
please test this

**Comment by hkumarmk** [comment]
@bodepd @sorenh could you please review it.

**Comment by bodepd** [comment]
This appears to only work when there is only one user specified. Adding another user results in the following error when trying to use sudo:

````
>>> /etc/sudoers.d/10_hkumar: Alias `READONLY' already defined near line 2 <<<
sudo: parse error in /etc/sudoers.d/10_hkumar near line 2
sudo: no valid sudoers sources found, quitting
````

It seems that the variable assigned needs to be uniquely named across all sudo.d files.

**Comment by hkumarmk** [comment]
Ah okay... ill just fix it

**Comment by hkumarmk** [comment]
@bodepd Done, fixed by appending username to command alias.

**Comment by hkumarmk** [comment]
please test this

**Comment by bodepd** [comment]
visudo validation is failing, I'm looking into it.

File: manifests/system/accounts/sudo/conf.pp [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  ##
+  # empty array in commands allowed means all commands are allowed (isn't it okay?)
+  ##
+  if empty($commands_allowed) {
+    $sudo_conf = "#Managed By Puppet\n${user} ALL=(ALL) NOPASSWD: ALL"
+  } else {
+    $commands_allowed_list = join($commands_allowed,',')
+
+    $sudo_conf = "#Managed By Puppet
+Cmnd_Alias 'CMND_${cmdalias_name_uc}' = ${commands_allowed_list}
```
Comment (bodepd) [inline manifests/system/accounts/sudo/conf.pp]: the '' around CMND_${cmdalias_name_uc} cause the resulting sudo file to be invalid, if you remove them it seems to work.

**Comment by bodepd** [comment]
rebased to clean up history and made the required modification as #661

**Pull request closed without merging**